site/docs/skylark/language.md

```diff
@@ -37,142 +29,97 @@ def fizz_buzz(n):
 fizz_buzz(20)
 ```
 
-The following basic types are supported: [None](lib/globals.html#None),
-[bool](lib/bool.html), [dict](lib/dict.html), function, [int](lib/int.html),
-[list](lib/list.html), [string](lib/string.html). On top of that, two new
-types are specific to Bazel: [depset](lib/depset.html) and
-[struct](lib/struct.html).
-
-Starlark is syntactically a subset of Python 3, and will remain so through at
-least the 1.x release lifecycle. This ensures that Python-based tooling can at
-least parse Starlark code. Although Starlark is not *semantically* a subset of
-Python, behavioral differences are rare (excluding cases where Starlark raises
-an error).
+Starlark's semmantics can differ from Python, and behavioral differences are rare, except for cases where Starlark raises an error. The following Python types are supported:
 
+* [None](lib/globals.html#None)
+* [bool](lib/bool.html)
+* [dict](lib/dict.html)
+* function
+* [int](lib/int.html)
+* [list](lib/list.html)
+* [string](lib/string.html)
 
 ## Mutability
 
-Because evaluation of BUILD and .bzl files is performed in parallel, there are
-some restrictions in order to guarantee thread-safety and determinism. Two
-mutable data structures are available: [lists](lib/list.html) and
-[dicts](lib/dict.html).
+Starlark favors immutability. Two mutable data structures are available: [lists](lib/list.html) and [dicts](lib/dict.html). Changes to mutable data-structures, such as appending a value to a list or deleting an entry in a dictionary are valid only for objects created in the current context. After a context finishes, its values become immutable.
 
-In a build, there are many "evaluation contexts": each `.bzl` file and each
-`BUILD` file is loaded in a different context. Each rule is also analyzed in a
-separate context. We allow side-effects (e.g. appending a value to a list or
-deleting an entry in a dictionary) only on objects created during the current
-evaluation context. Once the code in that context is done executing, all of its
-values are frozen.
+This is because Bazel builds use parallel execution. During a build, each `.bzl` file and each `BUILD` file get their own execution context. Each rule is also analyzed in its own context.
 
-For example, here is the content of the file `foo.bzl`:
+Let's go through an example. Let the below be the content of the file `foo.bzl`:
```

Comment:
This sounds weird to me: "Let the below be the content" (but I'm not native speaker)